bin/generate-tiles

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# For backward compatibility, allow both PG* and POSTGRES_* forms,
+# with the non-standard POSTGRES_* form taking precedence.
+# An error will be raised if neither form is given, except for the PGPORT
+export PGHOST="${POSTGRES_HOST:-${PGHOST?}}"
+export PGDATABASE="${POSTGRES_DB:-${PGDATABASE?}}"
+export PGUSER="${POSTGRES_USER:-${PGUSER?}}"
+export PGPASSWORD="${POSTGRES_PASSWORD:-${PGPASSWORD?}}"
+export PGPORT="${POSTGRES_PORT:-${PGPORT:-5432}}"
+
+# List of postgres servers
+if [[ -z "${PGHOSTS_LIST}" ]] # "xxx.xxx.xxx.xxx&host=xxx.xxx.xxx.xxx&host=..."
+then
+  export HOST_COUNT=1
+  export PGHOSTS="${PGHOST}"
+else
+  export HOST_COUNT=`awk -F"&" '{print NF}' <<< "${PGHOSTS_LIST}"`
+  export PGHOSTS="${PGHOSTS_LIST}"
+fi
+
+export NO_GZIP=${NO_GZIP:-1}
+if [[ ${NO_GZIP} ]] && [[ ! ${NO_GZIP} == "0" ]] && [[ ! ${NO_GZIP} == "true" ]]
+then
+  export NO_GZIP=1
+fi
```

Comment:
AFAIK, you should never use vars like this without quotes.  Something like this perhaps?  And same for the KEY (btw, we may want to rename KEY -- too confusing -- how about `USE_KEY` or `USE_KEY_COLUMN` ?

```suggestion
if [[ -n "${NO_GZIP:}" ]] && [[ "${NO_GZIP:}" != "0" ]] && [[ "${NO_GZIP:}" != "false" ]]; then
  export NO_GZIP=1
else
  export NO_GZIP=
fi
```

Reply:
+1 for `USE_KEY_COLUMN`. It seems to me more descriptive.

Reply:
@nyurik just not sure what is the purpose of colon in `"${NO_GZIP:}`. When I test it I get `bash: ${NO_GZIP:}: bad substitution`

Reply:
It also seems to me that no matter what you have set for NO_GZIP it always get reset to either empty or 1. Even though you have it set for true or 0.

Reply:
@lazaa32 sorry, forgot a dash (fixed in the comment).  The `"${FOO:-}"` would use FOO unless its empty or null, in which case it will use whatever is after the `:-`. In this case, the value is an empty string.  Note that just using `${FOO}` would not work because of the `set -u` above, so you have to be explicit here.

Reply:
@nyurik thanks, I'm not a really expert in bash so I wondered what that syntax means :D . However it still seems to me that if  you set `NO_GZIP=0` after this condition you end up with `empty`. If you set `NO_GZIP=true` you end up with `NO_GZIP=1`. Am I wrong when i think that boolean works like this `false=1=empty` and `true=0`?

Reply:
The bash manual says:
> **${_parameter_:-_word_}**
     If _parameter_ is _unset_ or _null_, the expansion of _word_ is substituted. Otherwise, the value of _parameter_ is substituted.

Therefore "${FOO}` and `${FOO:-}` are identical

```suggestion
case "${NO_GZIP:-0}" in
  0 | false ) export NO_GZIP= ;;
  * ) export NO_GZIP=1 ;;
esac
```

Reply:
@zstadler this is not correct because `set -u` (same as `set -o nounset`) will trigger an error if you try to reference an undeclared variable.

Reply:
@lazaa32 I think the "cleanest solution" (whatever that even means)  would be to actually set `NO_GZIP` and other boolean vars to the final result. Something like this:

```if ...; then
  export NO_GZIP="&nogzip=1"
else
  export NO_GZIP=
fi

...

PGQUERY="...${NO_GZIP}..."
```

This way we create the exact part of the query as desired. This also means that there is no need to define the value for TEST_ON_STARTUP -- because you won't even specify that value if the user hasn't set it.  What do you think?

Reply:
@nyurik, you are right about `${FOO:-}`

I believe my suggestion makes this logic easier to read.

Reply:
TBH, i'm not in favor of the `case` statement -- much less readable than an `if` with a list of conditions. I'm not even sure if we use case anywhere else. Also, I think it would be better to just declare the whole `&nogzip=1` vs an empty value -- this way you control exactly what you expand.

Reply:
+1 to keep `if` since we don't use `case` in other scripts. Also exported the final string directly into var.